mongoengine/connection.py

```diff
@@ -1,17 +1,22 @@
 from pymongo import MongoClient, ReadPreference, uri_parser
-from mongoengine.python_support import (IS_PYMONGO_3, str_types)
+from mongoengine.python_support import (IS_PYMONGO_27, IS_PYMONGO_3, str_types)
 
 __all__ = ['ConnectionError', 'connect', 'register_connection',
            'DEFAULT_CONNECTION_NAME']
 
 
 DEFAULT_CONNECTION_NAME = 'default'
+AUTHENTICATION_MECHANISM = 'DEFAULT'
+
 if IS_PYMONGO_3:
     READ_PREFERENCE = ReadPreference.PRIMARY
 else:
     from pymongo import MongoReplicaSetClient
     READ_PREFERENCE = False
 
+if IS_PYMONGO_27:
```

Comment:
**mongoengine** is supported pymongo >= 2.7. Is not it?
for pymogo v2.7 is required authmechanism = 'MONGODB-CR' by default:

https://github.com/mongodb/mongo-python-driver/blob/v2.7/pymongo/mongo_client.py#L372

While all the other pymongo versions can use the "DEFAULT", It is pymongo requires:

https://github.com/mongodb/mongo-python-driver/blob/v2.8/pymongo/mongo_client.py#L381
https://github.com/mongodb/mongo-python-driver/blob/v2.9/pymongo/mongo_client.py#L431